framework/test/gtest/AnyMapTest.cpp

```diff
@@ -250,6 +250,83 @@ TEST(AnyMapTest, CIHashUnique)
   any_map::unordered_any_cimap::hasher hash;
   std::size_t hashV1 = hash(v1);
   std::size_t hashV2 = hash(v2);
-  
+
   ASSERT_EQ(true, hashV1 != hashV2);
 }
+
+TEST(AnyMapTest, GeneralUsage)
+{
+  ASSERT_THROW(AnyMap m(static_cast<AnyMap::map_type>(100)), std::logic_error);
+
+  AnyMap om(AnyMap::ORDERED_MAP);
+  ASSERT_EQ(0, om.size());
+  ASSERT_TRUE(om.empty());
+  ASSERT_EQ(0, om.count("key1"));
+
+  auto it =
+    om.insert(std::make_pair(std::string("key1"), Any(std::string("val1"))));
+  ASSERT_TRUE(it.second);
+  ASSERT_EQ("key1", it.first->first);
+  ASSERT_EQ(std::string("val1"), any_cast<std::string>(it.first->second));
+  ASSERT_EQ(std::string("val1"), any_cast<std::string>(om["key1"]));
+
+  AnyMap uoci(AnyMap::UNORDERED_MAP_CASEINSENSITIVE_KEYS);
+  uoci["FIRST"] = 1;
```

Comment:
in the original test, the "I" is lower case. I think this was intentional since this is a case insensitive match and there is a test checking for "FIRST" instead of "FiRST". I'd suggest maintaining this in ported gtest.